demos/common/processing/async_pipeline.cpp

```diff
@@ -0,0 +1,217 @@
+    slog::info << "Loading Inference Engine" << slog::endl;
+
+    slog::info << "Device info: " << slog::endl;
+    slog::info<< ie->GetVersions(cnnConfig.devices);
+
+    /** Load extensions for the plugin **/
+    if (!cnnConfig.cpuExtensionsPath.empty()) {
+        // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
+        IExtensionPtr extension_ptr = make_so_pointer<IExtension>(cnnConfig.cpuExtensionsPath.c_str());
+        ie->AddExtension(extension_ptr, "CPU");
+    }
+    if (!cnnConfig.clKernelsConfigPath.empty()) {
+        // clDNN Extensions are loaded from an .xml description and OpenCL kernel files
+        ie->SetConfig({ {PluginConfigParams::KEY_CONFIG_FILE, cnnConfig.clKernelsConfigPath} }, "GPU");
+    }
+
+    // --------------------------- 2. Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
+    slog::info << "Loading network files" << slog::endl;
+    /** Read network model **/
+    InferenceEngine::CNNNetwork cnnNetwork = ie->ReadNetwork(model->getModelFileName());
+    /** Set batch size to 1 **/
+    slog::info << "Batch size is forced to 1." << slog::endl;
+    cnnNetwork.setBatchSize(1);
+
+    // -------------------------- Reading all outputs names and customizing I/O blobs (in inherited classes)
+    model->prepareInputsOutputs(cnnNetwork);
+
+    // --------------------------- 4. Loading model to the device ------------------------------------------
+    slog::info << "Loading model to the device" << slog::endl;
+    execNetwork = ie->LoadNetwork(cnnNetwork, cnnConfig.devices, cnnConfig.execNetworkConfig);
+
+    // --------------------------- 5. Create infer requests ------------------------------------------------
+    requestsPool.reset(new RequestsPool(execNetwork, cnnConfig.maxAsyncRequests));
+
+    // --------------------------- 6. Call onLoadCompleted to complete initialization of model -------------
+    model->onLoadCompleted(&execNetwork, requestsPool.get());
+}
+
+PipelineBase::~PipelineBase(){
+    waitForTotalCompletion();
+}
+
+
+void PipelineBase::waitForData(bool shouldKeepOrder){
+    std::unique_lock<std::mutex> lock(mtx);
+
+    condVar.wait(lock, [&] {return callbackException != nullptr ||
+        requestsPool->isIdleRequestAvailable() ||
+        (shouldKeepOrder ?
+            completedInferenceResults.find(outputFrameId) != completedInferenceResults.end() :
+            !completedInferenceResults.empty());
+    });
+
+    if (callbackException)
+        std::rethrow_exception(callbackException);
+}
+
+int64_t PipelineBase::submitRequest(InferenceEngine::InferRequest::Ptr request, std::shared_ptr<MetaData> metaData){
+    perfInfo.numRequestsInUse = (uint32_t)requestsPool->getInUseRequestsCount();
+
+    auto frameStartTime = std::chrono::steady_clock::now();
+
+    if (!perfInfo.startTime.time_since_epoch().count())
```

Comment:
See https://github.com/openvinotoolkit/open_model_zoo/pull/1330#discussion_r461006352

Reply:
Please give example: in which case it won't work ?

Reply:
I don't have any

Reply:
so, let's leave it as is